File: server/DataCore/invwh.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace inv19.models { 
	/* invwh -  Структура склада */ 

 public class  invwh_loc { // Стеллаж
	 public System.Guid  invwh_locId{ get; set; } // Идентификатор (первичный ключ)
	 public List<invwh_cell>  invwh_cell { get; set; } // дочерний раздел: Ячейка
	public System.Guid  theStore { get; set; } //Склад
	
	[Required]
	public string  name{ get; set; } // Название
	public System.Guid  whZone { get; set; } //Зона
	
 }

 public class  invwh_cell { // Ячейка
	 public System.Guid  invwh_cellId{ get; set; } // Идентификатор (первичный ключ)
	[Required]
	 public System.Guid  invwh_locId { get; set; } // обратная ссылка на: Стеллаж
	[Required]
	public string  name{ get; set; } // Название
	public string  SHCODE{ get; set; } // Штрихкод ячейки
 }
}

File: client/src/app/invwh.ts
import { enums } from './enums';

export namespace invwh { 
	/* invwh -  Структура склада */ 

 export interface   invwh_loc { // Стеллаж
	invwh_locId:string; // Primary key field
	theStore:string; //Склад -> invd_store
	name:string; // Название
	whZone:string; //Зона -> invd_zone
	// add dereference fields 
	theStore_name :string; // dereference for invd_store
	whZone_name :string; // dereference for invd_zone
 }

 export interface   invwh_cell { // Ячейка
	invwh_cellId:string; // Primary key field
	  invwh_locId:string; // Стеллаж
	name:string; // Название
	sHCODE:string; // Штрихкод ячейки
 }
}